### PR TITLE
Changes to deprecate powervs-provider-id-fmt flat

### LINKED ISSUE #1024: Deprecate powervs-provider-id-fmt flag
/kind feature
/area provider/ibmcloud

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

With the CCM support for VPC via https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1015, We are introducing a generic flag named `provider-id-fmt` to set Provider ID format, We need deprecate the Power VS specific flag `powervs-provider-id-fmt` and use   `provider-id-fmt` instead.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by Karthik-K-N**
**What this PR does / why we need it**:

Currently for Power VS clusters provider id format is set via powervs-provider-id-fmt flag, With recent merge of VPC CCM support [PR](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1015) we have introduced flag `provider-id-fmt`, same can be used to set provider id format for Power VS as well.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1024

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: docs/proposal/20230113-capibm-cli-tool.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
+# CLI tool for Cluster API Provider IBM Cloud
```
Comment (mkumatag) [inline docs/proposal/20230113-capibm-cli-tool.md:1]: Why this doc included in this PR?

File: main.go [hunk 2]
```diff
@@ -177,7 +177,7 @@ func initFlags(fs *pflag.FlagSet) {
 		&options.ProviderIDFormat,
 		"provider-id-fmt",
 		string(options.ProviderIDFormatV1),
-		"ProviderID format is used set the Provider ID format for Machine (Currently for VPC machines only)",
+		"ProviderID format is used set the Provider ID format for Machine",
 	)
```
Comment (mkumatag) [inline main.go:181]: Lets add following code to deprecate it
```golang
	fs.MarkDeprecated("powervs-provider-id-fmt", "please use provider-id-fmt flag")
```

File: main.go [hunk 3]
```diff
@@ -191,17 +191,17 @@ func initFlags(fs *pflag.FlagSet) {
 func validateFlags() error {
 	switch options.ProviderIDFormatType(options.PowerVSProviderIDFormat) {
 	case options.PowerVSProviderIDFormatV1:
-		setupLog.Info("Using v1 version of Power VS ProviderID format")
+		setupLog.Info("Power VS ProviderID format is set to v1 version, Note: its deprecated use provider-id-fmt to provider id format")
```
Comment (mkumatag) [inline main.go:194]: with the above `MarkDeprecated` call, you may not need this explicit message anymore

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: docs/proposal/20230113-capibm-cli-tool.md (hunk 1 above)
Comment (Karthik-K-N) [reply docs/proposal/20230113-capibm-cli-tool.md]: By mistake removed it.

File: main.go (hunk 2 above)
Comment (Karthik-K-N) [reply main.go:181]: Updated it accordingly.

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**